packages/library/src/rpc.ts

```diff
@@ -1,12 +1,28 @@
-import { createSolanaRpcApi, SolanaRpcMethods } from '@solana/rpc-core';
-import { createJsonRpc } from '@solana/rpc-transport';
-import type { Rpc } from '@solana/rpc-transport/dist/types/json-rpc-types';
+import {
+    createSolanaRpcApi,
+    createSolanaRpcSubscriptionsApi,
+    SolanaRpcMethods,
+    SolanaRpcSubscriptions,
+} from '@solana/rpc-core';
+import { createJsonRpc, createJsonSubscriptionRpc } from '@solana/rpc-transport';
+import type { Rpc, RpcSubscriptions } from '@solana/rpc-transport/dist/types/json-rpc-types';
 
 import { DEFAULT_RPC_CONFIG } from './rpc-default-config';
 
 export function createSolanaRpc(config: Omit<Parameters<typeof createJsonRpc>[0], 'api'>): Rpc<SolanaRpcMethods> {
-    return createJsonRpc<SolanaRpcMethods>({
+    return createJsonRpc({
         ...config,
         api: createSolanaRpcApi(DEFAULT_RPC_CONFIG),
     });
 }
+
+export function createSolanaRpcSubscriptions(
+    config: Omit<Parameters<typeof createJsonSubscriptionRpc>[0], 'api'>
+): RpcSubscriptions<SolanaRpcSubscriptions> {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-expect-error
+    return createJsonSubscriptionRpc({
+        ...config,
+        api: createSolanaRpcSubscriptionsApi(DEFAULT_RPC_CONFIG),
+    });
```

Comment:
What's the type error here?

Reply:
Oh, it goes away in the very next PR where I actually implement the first RPC typespec.

Reply:
Spotted that, ty